pulsar-client/src/main/java/org/apache/pulsar/client/impl/MemoryLimitController.java

```diff
@@ -47,6 +47,16 @@ public MemoryLimitController(long memoryLimitBytes, long triggerThreshold, Runna
 
     public void forceReserveMemory(long size) {
         long newUsage = currentUsage.addAndGet(size);
+        if (size < 0 && newUsage - size > memoryLimit
```

Comment:
does `forceReserveMemory` allow negative values?

Reply:
I'd suggest that we add an assert here and instead fix the logic in https://github.com/apache/pulsar/blob/aba50f2a276412bc43a4652b9ce303d384f71966/pulsar-client/src/main/java/org/apache/pulsar/client/impl/BatchMessageContainerImpl.java#L329 to call forceReserveMemory(delta) if delta > 0 and call releaseMemory(-delta) if delta < 0. no call if delta == 0. WDYT?

Reply:
Good Point! I have addressed your comment. PTAL. Thanks.